src/components/cuda/cupti_profiler.c

```diff
@@ -0,0 +1,1564 @@
+            .pPriv = NULL,
+            .pChipName = avail_events[gpu_id].chip_name,
+        };
+        NVPW_CALL(NVPW_CUDA_MetricsContext_CreatePtr(&(avail_events[gpu_id].metricsContextCreateParams)),
+            goto fn_fail; );
+
+        NVPW_MetricsContext_GetMetricNames_Begin_Params getMetricNameBeginParams = {
+            .structSize = NVPW_MetricsContext_GetMetricNames_Begin_Params_STRUCT_SIZE,
+            .pPriv = NULL,
+            .pMetricsContext = avail_events[gpu_id].metricsContextCreateParams.pMetricsContext,
+            .hidePeakSubMetrics = !listsubmetrics,
+            .hidePerCycleSubMetrics = !listsubmetrics,
+            .hidePctOfPeakSubMetrics = !listsubmetrics,
+        };
+        NVPW_CALL(NVPW_MetricsContext_GetMetricNames_BeginPtr(&getMetricNameBeginParams), goto fn_fail; );
+
+        avail_events[gpu_id].num_metrics = getMetricNameBeginParams.numMetrics;
+        avail_events[gpu_id].nv_metrics = initialize_dynamic_event_list_size(avail_events[gpu_id].num_metrics);
+        if (avail_events[gpu_id].nv_metrics == NULL) {
+            papi_errno = PAPI_ENOMEM;
+            goto fn_exit;
+        }
+        for (i=0; i<avail_events[gpu_id].num_metrics; i++) {
+            papi_errno = insert_event_record(avail_events[gpu_id].nv_metrics,
+                                      getMetricNameBeginParams.ppMetricNames[i],
+                                      i, 0);
+            if (papi_errno != PAPI_OK)
+                goto fn_exit;
+        }
+
+        NVPW_MetricsContext_GetMetricNames_End_Params getMetricNameEndParams = {
+            .structSize = NVPW_MetricsContext_GetMetricNames_End_Params_STRUCT_SIZE,
+            .pPriv = NULL,
+            .pMetricsContext = avail_events[gpu_id].metricsContextCreateParams.pMetricsContext,
+        };
+        NVPW_CALL(NVPW_MetricsContext_GetMetricNames_EndPtr((NVPW_MetricsContext_GetMetricNames_End_Params *) &getMetricNameEndParams), goto fn_fail; );
+
+    }
+    char evt_name[PAPI_2MAX_STR_LEN];
+    ntv_event_t *find=NULL;
+    int len;
+    for (gpu_id=0; gpu_id<num_gpus; gpu_id++) {
+        for (i=0; i<avail_events[gpu_id].num_metrics; i++) {
+            len = snprintf(evt_name, PAPI_2MAX_STR_LEN, "%s:device=%d", avail_events[gpu_id].nv_metrics->evts[i].name, gpu_id);
+            if (len > PAPI_2MAX_STR_LEN) {
+                ERRDBG("String formatting exceeded maximum length.\n");
+                papi_errno = PAPI_ENOMEM;
+                goto fn_exit;
+            }
+            if (find_event_name(all_evt_names, evt_name, &find) == PAPI_ENOEVNT) {
+                papi_errno = insert_event_record(all_evt_names, evt_name, i, 0);
+                if (papi_errno != PAPI_OK) {
+                    goto fn_exit;
+                }
+            }
+        }
+    }
+fn_exit:
+    return papi_errno;
+fn_fail:
+    papi_errno = PAPI_EMISC;
+    goto fn_exit;
+}
+
+int cuptip_get_event_description(const char *evt_name, char *description)
```

Comment:
`event_name_to_descriptor`. I think I got this wrong too in the rocm component. I need to fix the naming too

Reply:
changed to `cuptip_event_name_to_descr`. see similar prev comment.